src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -461,39 +461,79 @@ private static bool ParsePackage(InstallerMetadata installerMetadata, Manifests
-
-            bool parseResult = ParseExeInstallerType(path, baseInstaller, newInstallers) ||
-                (parseMsixResult = ParseMsix(path, baseInstaller, manifests, newInstallers)) ||
-                ParseMsi(path, baseInstaller, manifests, newInstallers);
-
-            if (!parseMsixResult)
-            {
-                // Only capture architecture if installer is non-msix as the architecture for msix installers is deterministic
-                var urlArchitecture = installerMetadata.UrlArchitecture = GetArchFromUrl(baseInstaller.InstallerUrl);
-                installerMetadata.UrlArchitecture = GetArchFromUrl(baseInstaller.InstallerUrl);
-                installerMetadata.BinaryArchitecture = baseInstaller.Architecture;
-
-                var overrideArch = installerMetadata.OverrideArchitecture;
-
-                if (overrideArch.HasValue)
-                {
-                    baseInstaller.Architecture = overrideArch.Value;
-                }
-                else if (urlArchitecture.HasValue)
-                {
-                    baseInstaller.Architecture = urlArchitecture.Value;
-                }
-            }
-
-            return parseResult;
+            baseInstaller.InstallerSha256 = GetFileHash(packageFile);
+
+            List<string> installerPaths = new List<string>();
+
+            if (installerMetadata.IsZipFile)
+            {
+                baseInstaller.InstallerType = InstallerType.Zip;
+                List<string> relativeFilePaths = installerMetadata.RelativeFilePaths;
+
+                List<NestedInstallerFile> nestedInstallerFiles = new List<NestedInstallerFile>();
+                foreach (string relativeFilePath in relativeFilePaths)
+                {
+                    nestedInstallerFiles.Add(new NestedInstallerFile { RelativeFilePath = relativeFilePath });
+                    installerPaths.Add(Path.Combine(installerMetadata.ExtractedDirectory, relativeFilePath));
+                }
+
+                baseInstaller.NestedInstallerFiles = nestedInstallerFiles;
+            }
+            else
+            {
+                installerPaths.Add(packageFile);
+            }
+
+            bool parseResult = false;
+            bool parseMsixResult = false;
+
+            Architecture? nestedArchitecture = null;
+
+            // There will only be multiple installer paths if there are multiple nested portable installers in an zip archive.
+            foreach (string path in installerPaths)
+            {
+                bool parseExeResult = false;
+
+                parseResult = (parseExeResult = ParseExeInstallerType(path, baseInstaller, newInstallers)) ||
+                    (parseMsixResult = ParseMsix(path, baseInstaller, manifests, newInstallers)) ||
+                    ParseMsi(path, baseInstaller, manifests, newInstallers);
+
+                // Check if the detected architectures are the same for all nested portables (exe).
+                if (parseExeResult)

```

Comment:
> if (parseExeResult) [](http://example.com/codeflow?start=15&length=20)

this could be removed so it's generic enough for future multiple nested zip installers

Reply:
Removed if statement